app/src/main/java/com/mobileasone/dagger2workshop/data/network/NotesServiceApiImp.kt

```diff
@@ -2,22 +2,14 @@ package com.mobileasone.dagger2workshop.data.network
 
 import android.os.Handler
 import com.mobileasone.dagger2workshop.domain.Note
+import javax.inject.Inject
 
 class NotesServiceApiImpl
-private constructor() : NotesServiceApi {
+@Inject constructor() : NotesServiceApi {
```

Comment:
By using @Inject annotation in the constructors, Dagger will know the way to create objects of NotesServiceApiImpl. So we won't need to add the instantiation in a @Module class but it will be bound in the SampleAppModule with the instance provided as argument.